lib/cfn-nag/custom_rules/VpcHasFlowLogRule.rb

```diff
@@ -0,0 +1,35 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class VpcHasFlowLogRule < BaseRule
+  def rule_text
+    'VPC should have a flow log attached'
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W58'
+  end
+
+  def audit_impl(cfn_model)
+    violating_policies = cfn_model.resources_by_type('AWS::EC2::VPC')
+                                  .select do |vpc|
+      flowlog_for_vpc(cfn_model, vpc).nil?
+    end
+
+    violating_policies.map(&:logical_resource_id)
+  end
+
+  def flowlog_for_vpc(cfn_model, vpc)
+    cfn_model.resources_by_type('AWS::EC2::FlowLog').find do |flowlog|
+      if flowlog.resourceId && flowlog.resourceId['Ref']
+        flowlog.resourceId['Ref'] == vpc.logical_resource_id
```

Comment:
i think this works.... we could probably tighten this up by building a relationship in cfn-model and we need a more general Ref solution in the model but.... i think we can go with this.  only ask is that you add a test case for > 1 VPC in the sample template